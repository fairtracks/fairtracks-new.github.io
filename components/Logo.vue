<template>
  <nuxt-link
    to="/"
    contain
    height="100%"
    :aspect-ratio="1532 / 480"
    tag="img"
    :src="
      require('~/assets/illustrations/logos/FAIRtracks-logo-light-white.png')
    "
  />
</template>
