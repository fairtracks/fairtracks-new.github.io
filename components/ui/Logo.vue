<template>
  <nuxt-link
    to="/"
    contain
    height="100%"
    :aspect-ratio="1532 / 480"
    tag="img"
    :src="
      createAssetPath(
        'illustrations',
        'logos',
        'FAIRtracks-logo-light-white.png'
      )
    "
  />
</template>
