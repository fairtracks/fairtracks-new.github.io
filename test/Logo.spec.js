// TODO: Move common code to tests/setup.js ?

import Vue from 'vue'
import Vuetify from 'vuetify'

import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import Logo from '~/components/ui/Logo.vue'

Vue.use(Vuetify)

const factory = () => {
  return shallowMount(Logo, {
    stubs: {
      NuxtLink: RouterLinkStub,
    },
  })
}

describe('Logo', () => {
  test('is a Vue instance', () => {
    const wrapper = factory()
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders properly', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
