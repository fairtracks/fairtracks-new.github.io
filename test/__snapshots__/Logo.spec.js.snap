// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders properly 1`] = `
"<a style=\\"height: 100%;\\">
  <v-img-stub height=\\"100%\\" aspectratio=\\"3.191666666666667\\" alt=\\"FAIRtracks logo\\" contain=\\"true\\" options=\\"[object Object]\\" position=\\"center center\\" src=\\"/logos/FairTracks-logo-light-white.png\\" transition=\\"fade-transition\\"></v-img-stub>
</a>"
`;
